### PR TITLE
Adding gf_14 harden files

### PR DESCRIPTION
**Pull request by stdavids**

### CODE REVIEW

**Comment by taylor-bsg** [comment]
How sure are you about the settings for:

RET1N ( 1'b1   )                        \
          , .STOV  ( 1'b0   )                        \
          , .EMA   ( 3'b010 )                        \
          , .EMAW  ( 2'b00  )                        \
          , .EMAS

**Comment by stdavids** [comment]
Responded in separate email.

File: hard/gf_14/bsg_misc/bsg_gate_stack_gen.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+module bsg_'''+sys.argv[4],'''#(width_p="inv",harden_p=1)
+   ('''+"\n    ,".join(input_params)+'''
+    , output [width_p-1:0] o
+    );
+'''
+
+    for x in range(1,int(sys.argv[2])+1) :
+        print ''' if (harden_p && (width_p=='''+str(x)+'''))
+    begin:macro
+      bsg_rp_tsmc_40_'''+sys.argv[1]+'''_b'''+str(x)+''' gate(.*);
```
Comment (taylor-bsg) [inline hard/gf_14/bsg_misc/bsg_gate_stack_gen.py]: tsmc 40?

File: hard/gf_14/bsg_misc/bsg_mux_gen.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+# For example, for AOI22  A's are faster than B's and A0 is faster than A1.
+#
+
+fab = "gf_14"
+
+#nand2 = "ND2D2BWP #0 (.A1 (#1), .A2 (#2), .ZN(#3)                    );"
+#nand3 = "ND3D2BWP #0 (.A1 (#1), .A2 (#2), .A3 (#3), .ZN(#4)          );"
+#nand4 = "ND4D2BWP #0 (.A1 (#1), .A2 (#2), .A3 (#3), .A4 (#4), .ZN(#5));"
+
+nand2 = "SC7P5T_ND2X2_SSC14SL #0 (.A(#1), .B(#2), .Z(#3));"
```
Comment (taylor-bsg) [inline hard/gf_14/bsg_misc/bsg_mux_gen.py]: comment with justification (or lack of justification) for using high leakage cells

File: hard/gf_14/bsg_misc/bsg_reduce_gen.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/python
+
+import sys;
+
+print '''
+
+module bsg_rp_gf_14_reduce_and_b4 (input [3:0] i, output o);
```
Comment (taylor-bsg) [inline hard/gf_14/bsg_misc/bsg_reduce_gen.py]: comment with justification (or lack of justification) for using high leakage cells

**Comment by dpetrisko** [comment]
Is this PR still relevant after the tapeout branch that was merged?

**Comment by stdavids** [comment]
Nope, these changes are part of PR #111

**Pull request closed without merging**